internal/api/cluster_installation.go

```diff
@@ -120,6 +139,68 @@ func handleSetClusterInstallationConfig(c *Context, w http.ResponseWriter, r *ht
 		return
 	}
 
-	// TODO: write the config
-	w.WriteHeader(http.StatusNotImplemented)
+	cluster, err := c.Store.GetCluster(clusterInstallation.ClusterID)
+	if err != nil {
+		c.Logger.WithError(err).Error("failed to query cluster")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	if cluster == nil {
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	var applyConfig func(parentKey string, value map[string]interface{}) error
```

Comment:
Indeed -- I can't invoke the function recursively since it isn't defined at the time the body of the function is being parsed. (This isn't a problem for functions at the package level for the usual non-closure reasons.) Happy to hoist it out if that would seem cleaner.